File: NCAA/scripts/_calib.mjs
import fs from 'fs';
const h = JSON.parse(fs.readFileSync('data/history.json','utf8'));
const runs = h.runs || [];

// Check what sResult values look like
const resultVals = new Set();
let buckets = {};

for (const r of runs) {
  for (const g of (r.games || [])) {
    if (!g.sPick || g.sPick === 'PASS' || g.sResult === undefined) continue;
    resultVals.add(g.sResult);

    const p = g.pCover || 0;
    const b = Math.floor(p * 20) / 20;
    const key = b.toFixed(2);
    if (!buckets[key]) buckets[key] = { w: 0, l: 0, total: 0 };
    if (g.sResult === 'WIN' || g.sResult === 'W') buckets[key].w++;
    else if (g.sResult === 'LOSS' || g.sResult === 'L') buckets[key].l++;
    buckets[key].total++;
  }
}

console.log('sResult values found:', [...resultVals]);
console.log('\npCover bucket | W    | L    | Total | ActualRate | Expected');
const sorted = Object.entries(buckets).sort((a, b) => parseFloat(a[0]) - parseFloat(b[0]));
for (const [k, v] of sorted) {
  const n = v.w + v.l;
  const rate = n > 0 ? (v.w / n * 100).toFixed(1) : 'N/A';
  const expected = (parseFloat(k) * 100 + 2.5).toFixed(0); // midpoint
  const delta = n > 0 ? (v.w / n * 100 - parseFloat(k) * 100 - 2.5).toFixed(1) : 'N/A';
  console.log(`${k.padEnd(14)}| ${String(v.w).padEnd(5)}| ${String(v.l).padEnd(5)}| ${String(v.total).padEnd(6)}| ${rate}%`.padEnd(55) + `| ~${expected}%  Δ=${delta}`);
}

// Now show the calibration.mjs output
console.log('\n=== Calibration module output ===');
import { buildCalibrationTable } from './calibration.mjs';
const table = buildCalibrationTable(h);
for (const r of table) {
  const mid = Math.round(r.midpoint * 100);
  const sDelta = r.spread.winPct != null ? r.spread.winPct - mid : null;
  console.log(`${r.label.padEnd(10)} Spread: n=${String(r.spread.n).padEnd(5)} actual=${r.spread.winPct}%  expected=${mid}%  Δ=${sDelta}  units=${r.spread.units}`);
}

File: pyNCAA/scripts/_calib.py
#!/usr/bin/env python3
# scripts/_calib.py — Calibration check
import json
import math
from calibration import build_calibration_table

h = json.loads(open("data/history.json").read())
runs = h.get("runs") or []

result_vals = set()
buckets = {}

for r in runs:
    for g in (r.get("games") or []):
        if not g.get("sPick") or g["sPick"] == "PASS" or g.get("sResult") is None:
            continue
        result_vals.add(g["sResult"])
        p = g.get("pCover", 0)
        b = math.floor(p * 20) / 20
        key = f"{b:.2f}"
        if key not in buckets:
            buckets[key] = {"w": 0, "l": 0, "total": 0}
        if g["sResult"] in ("WIN", "W"):
            buckets[key]["w"] += 1
        elif g["sResult"] in ("LOSS", "L"):
            buckets[key]["l"] += 1
        buckets[key]["total"] += 1

print("sResult values found:", list(result_vals))
print("\npCover bucket | W    | L    | Total | ActualRate | Expected")
sorted_b = sorted(buckets.items(), key=lambda x: float(x[0]))
for k, v in sorted_b:
    n = v["w"] + v["l"]
    rate = f"{v['w'] / n * 100:.1f}" if n > 0 else "N/A"
    expected = f"{float(k) * 100 + 2.5:.0f}"
    delta = f"{v['w'] / n * 100 - float(k) * 100 - 2.5:.1f}" if n > 0 else "N/A"
    print(f"{k:<14}| {str(v['w']):<5}| {str(v['l']):<5}| {str(v['total']):<6}| {rate}%".ljust(55) + f"| ~{expected}%  D={delta}")

print("\n=== Calibration module output ===")
table = build_calibration_table(h)
if table:
    for r in table:
        mid = round(r["midpoint"] * 100)
        s_delta = r["spread"]["winPct"] - mid if r["spread"].get("winPct") is not None else None
        print(f"{r['label']:<10} Spread: n={str(r['spread']['n']):<5} actual={r['spread']['winPct']}%  expected={mid}%  D={s_delta}  units={r['spread']['units']}")
